pkg/remote/fileshare/s3fs/s3fs.go

```
@@ -31,94 +53,784 @@ func NewS3Client(config *aws.Config) *S3Client {
+		CanMkdir:  false,
+	}
+}
+
+func getParentPath(conn *connparse.Connection) string {
+	hostAndPath := conn.GetPathWithHost()
+	return getParentPathString(hostAndPath)
+}
+
+func getParentPathString(hostAndPath string) string {
+	parentPath := ""
+	slashIndices := slashRe.FindAllStringIndex(hostAndPath, 0)
```

golangci-lint (1.62.2)

SA1010: calling a FindAll method with n == 0 will return no results, did you mean -1?

(staticcheck)